### PR TITLE
apt-dater 1.0.4

### PR DESCRIPTION
**Pull request by alebcay**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/apt-dater.rb [hunk 1]
```diff
@@ -14,12 +14,15 @@ class AptDater < Formula
...
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
   depends_on "pkg-config" => :build
   depends_on "gettext"
   depends_on "glib"
   depends_on "popt"
 
   def install
+    system "autoreconf", "-ivf"
```
Comment (mistydemeo) [inline Formula/apt-dater.rb]: Does the prebuilt configure script not work in this release?

Comment (alebcay) [reply Formula/apt-dater.rb]: No, it errors out about missing `install-sh` and things like that. There is a `install-sh` provided in the download, but it turns out just to be a broken symlink. Running `autoreconf` seemed to set everything straight.

**Comment by fxcoudert** [comment]
Thanks @alebcay for this and your other PRs!

**Comment by MikeLund** [comment]
@alebcay Hello! Are you an actual user of this software, and have tested its functionality? As I've tried it just now, and even version 1.0.4 is not working properly in Homebrew, unlike the previous 0.9.0.

Please [see my previous comments](https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+apt-dater+is%3Aclosed) about this (and [some other user's](https://github.com/Homebrew/homebrew-core/pull/8924#issuecomment-273066762) as well for example), and consider rolling back this commit. There is no benefit to upgrading to 1.0.4 at this time and is only harmful.

Thanks for your time and consideration.

**Comment by fxcoudert** [comment]
Let me try to summarise:

- There were issues with 1.0.3 that lead Homebrew to downgrade back to 0.9.0
- 1.0.4 is indicated as the latest version, and marked stable, by upstream: https://github.com/DE-IBH/apt-dater/releases
- It's not clear what exactly the issues (with either 1.0.3 or 1.0.4) are, whether they have been reported upstream, and what is the status there. @MikeLund a comment on that would be welcome.

Regarding Homebrew's policies:

- We are not going to ship forever an older version of this project, that is not maintained anymore. So rolling back does not appear a viable option.
- The ideal case is that the bug is reported and fixed, and we can carry a patched version until the next release.
- Otherwise, if the recent versions of apt-dater are effectively unsupported on macOS, then the solution is to remove it from Homebrew.

**Comment by MikeLund** [comment]
> 1.0.4 is [..] marked stable

Kinda, small nitpick is that the [Changelog](https://github.com/DE-IBH/apt-dater/blob/master/ChangeLog#L1) has it marked as unstable, even if he didn't tag it as a prerelease in Github. It's also a hobby-project.

Anyways, some complaints from the previous issue is that changes aren't documented (their readme still only has the 0.x config format for example), and reverse-compatibility is completely broken.

But if we ignore that and become bleeding-edge, the **actual** issue is partially explained here: https://github.com/Homebrew/homebrew-core/issues/5762 -- basically, it _should_ read configs from `~/.config/apt-dater/` (and when you run the program for the first time, it does create these files), but instead it's only really reading configs from `/usr/local/Cellar/apt-dater/1.0.4/etc/apt-dater/`. As such: editing the configs in ~/.config doesn't do anything -- and if you decide to make changes directly into the Cellar (which is very naughty behavior), it will be lost if/when a new update happens.


I am pretty sure it's a Homebrew issue (there is no issues with the behavior in Debian for example), but I tried for a long time to resolve it and didn't get anywhere. And I don't expect apt-dater's dev to have an answer, so there's no bug filed for him.

So, if you had an idea to resolve the ~/.config issue, I would be happy enough with this newest version. Otherwise, it doesn't benefit anyone to push the newer version. Considering this (IMO major) issue, and there's no real features or critical fixes added in the 1.x release, it's my opinion that 0.9.0 benefits Homebrew users more, and that was the consensus last year. But if you had a packaging fix for 1.x, that would be wonderful. Thanks for reading.

**Comment by alebcay** [comment]
Upon some preliminary digging, I noticed that [the config check (creating directory/files if missing) determines the path with `g_getenv("HOME")`](https://github.com/DE-IBH/apt-dater/blob/master/src/keyfiles.c#L84):

```
 if(g_spawn_sync(g_getenv ("HOME"), argv, NULL, G_SPAWN_CHILD_INHERITS_STDIN, NULL, NULL, NULL, NULL, NULL, &error) == FALSE) {
```

but [the logic that handles actually loading configuration gets the config location with `g_get_user_config_dir()`](https://github.com/DE-IBH/apt-dater/blob/master/src/main.c#L85):

```
cfgdirname = g_strdup_printf("%s/%s", g_get_user_config_dir(), PACKAGE);
```

I wonder if switching that out for `g_getenv("HOME") + "/.config/apt-dater/"` (or some C equivalent, as this written as-is is probably incorrect syntax) might fix something.

**Comment by alebcay** [comment]
> Kinda, small nitpick is that the Changelog has it marked as unstable, even if he didn't tag it as a prerelease in Github. It's also a hobby-project.

By that changelog, 0.9.0 is also unstable. We should roll back to 0.8.6, the last "stable" version. After installation, I ran `apt-dater` once. As expected, it created the folder and files in `~/.config/apt-dater`, printing so on the terminal. I then ran it again, and as expected, it did not complain about missing files in `~/.config/apt-dater`. Moreover, I deleted `/usr/local/Cellar/apt-dater/1.0.4/etc/apt-dater` and ran `apt-dater` again and the program did not complain. The deleted folder in the Cellar also was not recreated in this run of the program.

I also did an `inreplace` on the build process to print `cfgfilename` and found that `g_get_user_config_dir()` was resolving to `/Users/<username>/.config/apt-dater`, as expected. So, as far as I can tell, I can't reproduce the issue.

**Comment by fxcoudert** [comment]
- All versions are marked unstable (as in Debian unstable, I think), and if we went by this criterion we'd remove it entirely.
- I don't expect this to be a Homebrew-specific bug, given we only run `configure` and `make` with standard arguments.

Closing, please open an upstream issue with full information to debug with them.

**Comment by MikeLund** [comment]
> Moreover, I deleted `/usr/local/Cellar/apt-dater/1.0.4/etc/apt-dater` and ran `apt-dater` again and the program did not complain.

Aha, and that's exactly what fixes it!

The configs in /etc (in Homebrew's case, `/usr/local/etc/apt-dater/hosts.xml, symlinked to ../../Cellar/apt-dater/1.0.4/etc/apt-dater/hosts.xml`) take precedence over what's inside your `~/.config`.

This can be expected behavior Linux (the permissions of these files in /etc can only be read by root), and so your ~/.config works properly... but Homebrew works a bit different - as it works pretty much fully in userspace.

The formula would be "helped" with a `"--sysconfdir=#{etc}"` (so the global config file would be persistent after an upgrade, instead of linking into the Cellar)... But the real solution would be to delete the global config dir after install, so the auto-created ~/.config will always take precedence. Is this possible?

**Comment by alebcay** [comment]
I think we can just manage to delete the `etc` subdirectory after build. The program doesn't even use the config in that directory. Ideally this should be fixed upstream, but until somebody submits it upstream, I guess I can try to submit a patch here. No guarantees on things getting accepted though.

**Comment by alebcay** [comment]
@MikeLund, this should be resolved in https://github.com/Homebrew/homebrew-core/commit/6bbdf684cb4270d301d3d40c95e470d6a7b8295f now.

**Comment by MikeLund** [comment]
Even after reinstalling after that update, I had to manually delete /usr/local/etc/apt-dater/ for some reason, but that's probably something I did wrong. So all is much better now, thank you!

**Pull request closed without merging**